tp: very large stack frames in trajectory planner

While looking at the output from the RTAI build I saw these:
```
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c: In function ‘tpHandleBlendArc.isra’:
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c:2136:1: warning: the frame size of 6528 bytes is larger than 2048 bytes [-Wframe-larger-than=]
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c: In function ‘tpAddCircle’:
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c:2302:1: warning: the frame size of 5584 bytes is larger than 2048 bytes [-Wframe-larger-than=]
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c: In function ‘tpAddLine’:
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c:2207:1: warning: the frame size of 5584 bytes is larger than 2048 bytes [-Wframe-larger-than=]
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c: In function ‘tpAddRigidTap’:
/home/runner/work/linuxcnc/linuxcnc/src/emc/tp/tp.c:1696:1: warning: the frame size of 5584 bytes is larger than 2048 bytes [-Wframe-larger-than=]
```
 A 5..6kB stack frame looks a bit overkill. Part of it may have been introduced with the new TP, but I'm not sure.
@grandixximo, maybe you know more about this? You have been in this code before. Are these stack frames required to be so large?

--- a/src/emc/tp/tp.c
+++ b/src/emc/tp/tp.c
@@ -1011,7 +1011,8 @@ STATIC tc_blend_type_t tpChooseBestBlend(TP_STRUCT const * const tp,
 }
 
 
-STATIC tp_err_t tpCreateLineArcBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc, TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
+STATIC __attribute__((__noinline__))
+tp_err_t tpCreateLineArcBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc, TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
 {
     tp_debug_print("-- Starting LineArc blend arc --\n");
 
@@ -1169,7 +1170,8 @@ STATIC tp_err_t tpCreateLineArcBlend(TP_STRUCT * const tp, TC_STRUCT * const pre
 }
 
 
-STATIC tp_err_t tpCreateArcLineBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc, TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
+STATIC __attribute__((__noinline__))
+tp_err_t tpCreateArcLineBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc, TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
 {
 
     tp_debug_print("-- Starting ArcLine blend arc --\n");
@@ -1310,7 +1312,8 @@ STATIC tp_err_t tpCreateArcLineBlend(TP_STRUCT * const tp, TC_STRUCT * const pre
     return TP_ERR_OK;
 }
 
-STATIC tp_err_t tpCreateArcArcBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc, TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
+STATIC __attribute__((__noinline__))
+tp_err_t tpCreateArcArcBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc, TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
 {
 
     tp_debug_print("-- Starting ArcArc blend arc --\n");
@@ -1479,7 +1482,8 @@ STATIC tp_err_t tpCreateArcArcBlend(TP_STRUCT * const tp, TC_STRUCT * const prev
 }
 
 
-STATIC tp_err_t tpCreateLineLineBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc,
+STATIC __attribute__((__noinline__))
+tp_err_t tpCreateLineLineBlend(TP_STRUCT * const tp, TC_STRUCT * const prev_tc,
         TC_STRUCT * const tc, TC_STRUCT * const blend_tc)
 {
 
@@ -2082,7 +2086,8 @@ static bool tpCreateBlendIfPossible(
  * blend arc. Essentially all of the blend arc functions are called through
  * here to isolate the process.
  */
-STATIC tc_blend_type_t tpHandleBlendArc(TP_STRUCT * const tp, TC_STRUCT * const tc) {
+STATIC __attribute__((__noinline__))
+tc_blend_type_t tpHandleBlendArc(TP_STRUCT * const tp, TC_STRUCT * const tc) {
 
     tp_debug_print("*****************************************\n** Handle Blend Arc **\n");
 

